lib/logmsg/logmsg.h

```diff
@@ -331,6 +331,26 @@ extern gint logmsg_node_max;
 LogMessage *log_msg_ref(LogMessage *m);
 void log_msg_unref(LogMessage *m);
 
+static inline NVTable *
+log_msg_get_payload(LogMessage *self)
+{
+  if (self->payload->borrowed)
+    {
+      log_msg_ref(self);
+      return NULL;
```

Comment:
I don't fully understand this API. get_payload suggests the payload will be returned, but instead I see NULL being returned and reference counts are increased at the same time.

I also noticed that we don't really use the payload where these functions are called.
My understanding is that we don't need the payload, we just want to keep things alive.

Can't we just do something like this?

```
LogMessagePin *log_msg_pin_payload(LogMessage *);
void log_msg_unpin_payload(LogMessagePin *);
```

`LogMessagePin` is just a marker, it can't be used by the caller.

Reply:
Good idea on the name, but I don't want to allocate anything, so it should just an opaque value.

Get/put are often used for taking and releasing a lock, that's the term I was trying to use, but didn't give it too much thought, and I agree it can be misleading.